pkg/cmd/init/utils.go

```diff
@@ -0,0 +1,195 @@
+	//create temporary directory to clone template into
+	dir, err := cmd.CreateTempDir(info.PathWorkingDir, ".template")
+	if err != nil {
+		logger.Debug("Error while creating temporary directory for clining template", zap.Error(err))
+		return utils.ErrorInternalServerError
+	}
+	defer func() {
+		_ = cmd.RemoveAll(dir)
+	}()
+
+	r, err := git.Clone(memory.NewStorage(), nil, &git.CloneOptions{URL: REPO})
+	if err != nil {
+		logger.Debug("Error while fetching templates from github", zap.Error(err))
+		return utils.ErrorFetchingTemplates
+	}
+
+	tags, err := r.Tags()
+	if err != nil {
+		logger.Debug("Error while getting github tags", zap.Error(err))
+		return msg.ErrorGetAllTags
+	}
+
+	tag, err := sortTag(tags, TemplateMajor)
+	if err != nil {
+		logger.Debug("Error while sorting tags for correct template application", zap.Error(err))
+		return msg.ErrorIterateAllTags
+	}
+
+	_, err = cmd.GitPlainClone(dir, false, &git.CloneOptions{
+		URL:           REPO,
+		ReferenceName: plumbing.ReferenceName(tag),
+	})
+	if err != nil {
+		logger.Debug("Error while fetching templates from github", zap.Error(err))
+		return utils.ErrorFetchingTemplates
+	}
+
+	azionDir := info.PathWorkingDir + "/azion"
+
+	// changing to Vulcan in case we are using any other type... this will be removed once Vulcan becomes the
+	// only adapter used by the cli
+	typeLang := info.Template
+	if typeLang != "nextjs" && typeLang != "static" && typeLang != "simple" {
+		typeLang = "vulcan"
+	}
+
+	//move contents from temporary directory into final destination
+	err = cmd.Rename(dir+"/webdev/"+typeLang, azionDir)
+	if err != nil {
+		logger.Debug("Error while copying files to current project directory", zap.Error(err))
+		return utils.ErrorMovingFiles
+	}
+
+	return nil
+}
+
+func (cmd *InitCmd) createJsonFile(options *contracts.AzionApplicationOptions, info *InitInfo) error {
+	data, err := json.MarshalIndent(options, "", "  ")
+	if err != nil {
+		return msg.ErrorUnmarshalAzionFile
+	}
+
+	err = cmd.WriteFile(info.PathWorkingDir+"/azion/azion.json", data, 0644)
+	if err != nil {
+		return utils.ErrorInternalServerError
+	}
+	return nil
+}
+
+func shouldFetch(cmd *InitCmd, info *InitInfo) (bool, error) {
+	var err error
+	var shouldFetchTemplates bool
+	if empty, _ := cmd.IsDirEmpty("./azion"); !empty {
+		if info.NoOption || info.YesOption {
+			shouldFetchTemplates = yesNoFlagToResponse(info)
+		} else {
+			prompt := &survey.Confirm{
+				Message: "This project was already configured. Do you want to override the previous configuration?",
+			}
+			err := survey.AskOne(prompt, &shouldFetchTemplates)
+			if err != nil {
+				return false, err
+			}
+		}
+
+		if shouldFetchTemplates {
+			err = cmd.CleanDir("./azion")
+			if err != nil {
+				logger.Debug("Error while trying to clean azion directory", zap.Error(err))
+				return false, err
+			}
+		}
+		return shouldFetchTemplates, nil
+	}
+	return true, nil
+}
+
+func askForInput(msg string, defaultIn string) (string, error) {
```

Comment:
I think you can add a struct input as input to make this func reusable? 
```suggestion
func askForInput(msg string, defaultIn string, prompt *survey.Input) (string, error) {
```